okhttp/src/test/java/okhttp3/ConscryptTest.java

```diff
@@ -31,23 +32,22 @@
 import static org.junit.Assert.assertTrue;
 
 public class ConscryptTest {
-  private OkHttpClient client = new OkHttpClient();
-
   @BeforeClass
-  public static void setup() {
-    assumeConscrypt();
-
-    OpenSSLProvider provider = new OpenSSLProvider();
-    Security.insertProviderAt(provider, 1);
+  public static void installProvider() {
+    if ("conscrypt".equals(System.getProperty("okhttp.platform"))) {
+      Security.insertProviderAt(Conscrypt.newProviderBuilder().provideTrustManager(true).build(), 1);
+    }
   }
 
+  private OkHttpClient client = new OkHttpClient();
+
   @After
   public void tearDown() {
     TestUtil.ensureAllConnectionsReleased(client);
   }
 
   private static void assumeConscrypt() {
-    Assume.assumeTrue("conscrypt".equals(System.getProperty("okhttp.platform")));
+    Assume.assumeTrue(Conscrypt.isConscrypt(Platform.get().platformTrustManager()));
```

Comment:
This is the honest check of whether we are running the test with Conscrypt enabled.